kernel_tuner/observers/ncu.py

```diff
@@ -0,0 +1,36 @@
+from kernel_tuner.observers import PrologueObserver
+
+try:
+    import nvmetrics
+except (ImportError):
+    nvmetrics = None
+    pass
+
+class NCUObserver(PrologueObserver):
+    """``NCUObserver`` measures performance counters.
+
+    """
+
+    def __init__(self, metrics=None):
+        """Create a new ``NCUObserver``.
+
+        :param metrics: The metrics to observe. This should be a list of strings.
+                        You can use ``ncu --query-metrics`` to get a list of valid metrics.
+        """
+
+        if not nvmetrics:
+            print("NCUObserver is not available.")
+
+        self.metrics = metrics
+        self.results = dict()
+
+    def before_start(self):
+        if nvmetrics:
```

Comment:
I think this is more of a Python problem, in case `nvmetrics` couldn't be imported, `nvmetrics` will be `None` and hence you cannot call any of the function. When raising an exception when `nvmetrics` could not be imported as you suggested above, this check can be removed at all.

Reply:
This doesn't seem relevant any more now that an exception is being raised.